BDArmory/Weapons/Missiles/MissileBase.cs

```diff
@@ -700,6 +731,24 @@ protected void UpdateRadarTarget()
 
             float angleToTarget = Vector3.Angle(radarTarget.predictedPosition - transform.position, GetForwardTransform());
 
+            if (hasIOG)
+            {
+                if (radarLOAL)
+                {
+                    maxRadarFailTime = 120;
+                }
+                else
+                {
+                    maxRadarFailTime = 30;
+                }
+                hasBI = false;
+            }
+
+            if(hasBI && !hasIOG)
+            {
+                maxRadarFailTime = 12;
+            }
+
```

Comment:
Not really understanding the point of this - having basic Inertial time out earlier - is the thought that you have a radar detect a target -> fires SARH radar missile at it via inertial guidance until missile enters active radar range -> missile IG times out before it enters active radar range and self-destructs? At that point, why are your missiles configured with a max range they can't actually reach?
Wouldn't it make more sense to have inertial timeout be based on some fraction of missile maxLaunchRange/optimalAirspeed? (Or use GPS w/radar terminal guidance, which already does this...) Else this is only going to work optimally with missiles with very specific speeds and launch ranges.

Reply:
the idea is of a basic analogue autopilot that fly the missile using inertia, for example AGM-114A or R-24R, and the IOG is a more advanced one that can acquire before launch data of the target to calculate where it need to go to intercept like AIM-120A, i still have other ideias to better implement the later advanced guidance but it's just a initial concept, one of my ideas were for exemple on AI vs AI one AI fires a 120A or a R-27R for example and the other defend using the ecm jammer, the 120A can acquire a new target during flight but if the target is inside his active radar range, case the target is too outside of it it will fly until the firing plane can get close enough to maintain a lock even with ecm, so the missile can reaquire data align and enter it's active radar state, in case of 27R it don't have an active radar , it's permit to reacquire a lock and fly to the target if the lock is maintained after that